crux_core/src/render.rs

```diff
@@ -1,27 +1,34 @@
 use super::Command;
-use crate::Capability;
+use crate::{capability::CapabilityContext, Capability};
 
-pub struct Render<Ef>
-where
-    Ef: Clone,
-{
-    make_effect: Ef,
+pub struct Render<Ev> {
+    context: CapabilityContext<(), Ev>,
 }
 
-impl<Ef> Render<Ef>
+impl<Ev> Render<Ev>
 where
-    Ef: Clone,
+    Ev: 'static,
 {
-    pub fn new(make_effect: Ef) -> Self {
-        Self { make_effect }
+    pub fn new(context: CapabilityContext<(), Ev>) -> Self {
+        Self { context }
     }
 
-    pub fn render<Ev>(&self) -> Command<Ef, Ev>
+    pub fn render(&self) {
+        let ctx = self.context.clone();
+        self.context
+            .spawn(async move { ctx.run_command(Command::new_without_callback(())) })
```

Comment:
Hmm, reading the other capabilities, I'm not sure why this runs a command (rather than emitting an effect), or what command is now.

Reply:
The `()` is just because `Command` has to package up some effect.  The render command doesn't have any payload so I just put a `()` in here.  The `CapabilityFactory` provided by the user handles mapping that into the `Effect::Render` the user provides.

`Command` still exists to ferry effects & a continuation callback up to the `Core`.  It's what `fn effect` is using under the hood.

But render was a bit of an odd capability that didn't have a callback - and the `effect` function always registers one.  For the sake of not changing it I just made this one run a command directly.  Atlhough could easily change it if we'd rather consistency?